test/query.spec.js

```diff
@@ -69,6 +70,30 @@ describe('Query', () => {
     })
   })
 
+  it('does not return an error if only some queries error', (done) => {
+    const peer = peerInfos[0]
+
+    // mock this so we can dial non existing peers
+    dht.switch.dial = (peer, callback) => callback()
+
+    let i = 0
+    const query = (p, cb) => {
+      if (i++ === 1) {
+        return cb(new Error('fail'))
+      }
+      cb(null, {
+        closerPeers: [peerInfos[2]]
+      })
+    }
+
+    const q = new Query(dht, peer.id.id, () => query)
+    q.run([peerInfos[1].id], (err, res) => {
+      expect(err).not.to.exist()
+      expect(res.finalSet.size).to.eql(2)
```

Comment:
Can we add a comment about why this is 2 and/or verify the contents of `finalSet`? It would be helpful to see and test exactly what we expect the final set to contain. The comments you added in the other tests are super helpful.